### PR TITLE
Add New Keycloak URL Reference

### PR DESCRIPTION
**Pull request by TommyJ1994**
## Motivation

Update the Auth SDK to use the top-level Keycloak Service URL field instead for authentication.

JIRA: https://issues.jboss.org/browse/AEROGEAR-2703

## Progress
- [x] Update the Keycloak URL
- [x] Update Tests/Build

### CODE REVIEW

**Comment by TommyJ1994** [comment]
@maleck13 @david-martin Any volunteers to verify this with the DMZ/Proxy use case? 

Just need to check:
- [ ] Auth Login/Logout works fine.
- [ ] There are no JWT errors in logcat - eg `Invalid JWT provided - .....`

**Comment by wtrocki** [comment]
@TommyJ1994 See: https://groups.google.com/forum/#!topic/aerogear/LBQKO0KERgQ

File: auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java [hunk 1]
```diff
@@ -36,7 +38,9 @@
     public KeycloakConfiguration(final ServiceConfiguration configuration) {
         nonNull(configuration, "configuration");
 
-        this.serverUrl = nonEmpty(configuration.getProperty(SERVER_URL_NAME), SERVER_URL_NAME);
+        this.authServerUrl = nonEmpty(configuration.getProperty(AUTH_SERVER_URL_NAME),
```
Comment (maleck13) [inline auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java]: We should always use the url field in the config and not rely on the auth-server-url

Comment (wei-lee) [reply auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java]: @TommyJ1994 Sorry I didn't make it clear in the JIRA. The "auth-server-url" field should not be used anywhere at all. Only the `url` field should be used.

Comment (TommyJ1994) [reply auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java]: To verify JWT tokens (`iss` field), you need the actual Keycloak URL, not the Proxy/DMZ one as it will fail, so the `auth-server-url` is only used for token verification, the `url` field is used for everything else. That make sense?

Comment (wei-lee) [reply auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java]: ah, right. OK, at long as it's not used for establish the communication with the server, then I think we are fine.

@TommyJ1994 can you add a comment to the code just to make it clear that the `auth-server-url` should only be used for token verification?

Comment (TommyJ1994) [reply auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java]: @wei-lee Yeah, sure.

**Comment by secondsun** [comment]
@TommyJ1994 Is proxy support in the mobile-cli or will I need to add the url field manually?

File: auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java [hunk 2]
```diff
@@ -90,6 +94,16 @@ public String getHostUrl() {
         return this.serverUrl;
     }
 
+    /**
+     * Get the Auth Server URL string of the Keycloak singleThreadService This URL should only be
+     * used to get the issuer field for token verification.
+     *
+     * @return the Auth Server URL of the Keycloak singleThreadService
```
Comment (wei-lee) [inline auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java]: remove `singleThreadService`

File: auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java [hunk 3]
```diff
@@ -90,6 +94,16 @@ public String getHostUrl() {
...
+     */
+    public String getAuthHostURL() {
```
Comment (wei-lee) [inline auth/src/main/java/org/aerogear/mobile/auth/configuration/KeycloakConfiguration.java]: If this is only used for issuer verification, would it better to rename it to `getIssuerHostname` to avoid confusion?

File: docs/getting-started/certificate-pinning-best-practices.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
+= Overview
```
Comment (wei-lee) [inline docs/getting-started/certificate-pinning-best-practices.adoc]: is this file required as part of this PR?

File: filelist.log [hunk 5]
```diff
@@ -0,0 +1,204 @@
+.circleci/config.yml
```
Comment (wei-lee) [inline filelist.log]: should this file be here?

File: docs/getting-started/certificate-pinning-best-practices.adoc (hunk 4 above)
Comment (TommyJ1994) [reply docs/getting-started/certificate-pinning-best-practices.adoc]: @wei-lee woops, no. I'll remove now.

**Comment by wei-lee** [comment]
@TommyJ1994 I think we can merge this
